projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/linked-cases/services/linked-cases.service.ts

```diff
@@ -0,0 +1,148 @@
+              private readonly searchService: SearchService) {
+    this.jurisdictionService.getJurisdictions().subscribe((jurisdictions) => {
+      this.jurisdictionsResponse = jurisdictions;
+    }, (error) => {
+      this.serverJurisdictionError = true;
+    });
+  }
+
+  public groupLinkedCasesByCaseType = (arrObj, key) => {
+    return arrObj.reduce((rv, x) => {
+      (rv[x.value[key]] = rv[x.value[key]] || []).push(x.value['CaseReference']);
+      return rv;
+    }, {});
+  };
+
+  public constructElasticSearchQuery(caseIds: any[], size: number): ESQueryType {
+    return {
+      query: {
+        terms: {
+          reference: caseIds,
+        },
+      },
+      size,
+    };
+  }
+
+  public mapResponse(esSearchCasesResponse) {
+    const caseInfo = this.caseFieldValue.find(item => item.value && item.value.CaseReference === esSearchCasesResponse.case_id);
+    return caseInfo && {
+      caseReference: esSearchCasesResponse.case_id,
+      caseName: esSearchCasesResponse.case_fields.caseNameHmctsInternal || LinkedCasesService.CASE_NAME_MISSING_TEXT,
+      caseType: this.mapLookupIDToValueFromJurisdictions('CASE_TYPE', esSearchCasesResponse.case_fields['[CASE_TYPE]']),
+      service: this.mapLookupIDToValueFromJurisdictions('JURISDICTION', esSearchCasesResponse.case_fields['[JURISDICTION]']),
+      state: this.mapLookupIDToValueFromJurisdictions('STATE', esSearchCasesResponse.case_fields['[STATE]']),
+      reasons: caseInfo.value && caseInfo.value.ReasonForLink &&
+        caseInfo.value.ReasonForLink.map(reason => reason.value && reason.value.Reason),
+    }
+  }
+
+  public searchCasesByCaseIds(searchCasesResponse: any[]) {
+    return forkJoin(searchCasesResponse);
+  }
+
+  public getAllLinkedCaseInformation() {
+    const searchCasesResponse = [];
+    const linkedCaseIds = this.groupLinkedCasesByCaseType(this.caseFieldValue, 'CaseType');
+    Object.keys(linkedCaseIds).forEach(key => {
+      const esQuery = this.constructElasticSearchQuery(linkedCaseIds[key], 100)
+      const query = this.searchService.searchCasesByIds(key, esQuery, SearchService.VIEW_WORKBASKET);
+      searchCasesResponse.push(query);
+    });
+    if (searchCasesResponse.length) {
+      this.searchCasesByCaseIds(searchCasesResponse).subscribe((searchCases: any) => {
+        const casesResponse = [];
+        searchCases.forEach(response => {
+          response.results.forEach((result: any) => {
+            casesResponse.push(this.mapResponse(result));
+          });
+        });
+
+        const caseLinks = casesResponse.map(item => {
+          return {
+            caseReference: item.caseReference,
+            caseName: item.caseName,
+            caseService: item.service,
+            caseType: item.caseType,
+            unlink: false,
+            reasons: item.reasons && item.reasons.map(reason => {
+              return {
+                reasonCode: reason
+              } as LinkReason
```

Comment:
@johnbenjamin-hmcts - missing semicolon

Reply:
@Daniel-Lam I have updated the code now.